paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/PaymentOption.kt

```diff
@@ -1,20 +1,224 @@
 package com.stripe.android.paymentsheet.model
 
+import android.content.res.ColorStateList
+import android.content.res.Resources
+import android.graphics.BlendMode
+import android.graphics.Canvas
+import android.graphics.ColorFilter
+import android.graphics.Insets
+import android.graphics.Outline
+import android.graphics.PorterDuff
+import android.graphics.Rect
+import android.graphics.Region
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.ShapeDrawable
+import android.os.Build
 import androidx.annotation.DrawableRes
+import androidx.annotation.RequiresApi
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 /**
  * The customer's selected payment option.
  */
-data class PaymentOption(
+data class PaymentOption
+@Deprecated("Not intended for public use.")
+constructor(
     /**
      * The drawable resource id of the icon that represents the payment option.
      */
-    @DrawableRes val drawableResourceId: Int,
+    @Deprecated("Please use fetchIcon instead.")
+    @DrawableRes
+    val drawableResourceId: Int,
 
     /**
      * A label that describes the payment option.
      *
      * For example, "····4242" for a Visa ending in 4242.
      */
     val label: String
-)
+) {
+    // These aren't part of the primary constructor in order to maintain binary compatibility.
+    internal var lightThemeIconUrl: String? = null
+        private set
+
+    internal var darkThemeIconUrl: String? = null
+        private set
+
+    private var imageLoader: suspend (PaymentOption) -> Drawable = {
+        throw IllegalStateException("Must pass in an image loader to use iconDrawable.")
+    }
+
+    @Suppress("DEPRECATION")
+    internal constructor(
+        @DrawableRes drawableResourceId: Int,
+        label: String,
+        lightThemeIconUrl: String?,
+        darkThemeIconUrl: String?,
+        imageLoader: suspend (PaymentOption) -> Drawable
+    ) : this(drawableResourceId, label) {
+        this.lightThemeIconUrl = lightThemeIconUrl
+        this.darkThemeIconUrl = darkThemeIconUrl
+        this.imageLoader = imageLoader
+    }
+
+    /**
+     * Fetches the icon associated with this [PaymentOption].
+     */
+    fun icon(): Drawable {
```

Comment:
This is new public API. Should be in the changelog.

Reply:
Good catch. I updated the Changelog with the suggested items.